src/OpenLoco/src/Input.h

```diff
@@ -36,42 +37,49 @@ namespace OpenLoco::Input
         giant = 1,
     };
 
-    namespace Flags
+    enum class Flags : uint32_t
     {
-        constexpr uint32_t widgetPressed = 1 << 0;
-        constexpr uint32_t flag1 = 1 << 1;
-        constexpr uint32_t flag2 = 1 << 2;
-        constexpr uint32_t toolActive = 1 << 3;
-        constexpr uint32_t flag4 = 1 << 4;
-        constexpr uint32_t flag5 = 1 << 5;
-        constexpr uint32_t flag6 = 1 << 6;
-        constexpr uint32_t viewportScrolling = 1 << 7;
-    }
-
-    namespace MapSelectionFlags
+        none = 0U,
+        widgetPressed = 1 << 0,
+        flag1 = 1 << 1,
+        flag2 = 1 << 2,
+        toolActive = 1 << 3,
+        flag4 = 1 << 4,
+        flag5 = 1 << 5,
+        flag6 = 1 << 6,
+        viewportScrolling = 1 << 7,
```

Comment:
Add also unsigned suffix to the rest, signed shift is not the same as unsigned shift, flags should always be unsigned.